brainiak/isfc.py

```diff
@@ -1,226 +1,1104 @@
+                                                         np.nan),
+                                                 np.full((groups[group_labels[1]],
+                                                          groups[group_labels[1]]),
+                                                         group_labels[1])))))
+            np.fill_diagonal(group_matrix, np.nan)
+            
+            # Unsort matrix and squareform to create selector
+            group_selector = squareform(group_matrix[unsorter, :][:, unsorter],
+                                        checks=False)
+            
+        # If leave-one-out approach, just user group assignment as selector
+        elif pairwise == False:
+            group_selector = group_assignment
+    
+    # Manage one-sample and incorrect group assignments
+    elif not group_assignment or len(np.unique(group_assignment)) == 1:
+        n_groups = 1
+        
+        # If pairwise initialize matrix of ones for sign-flipping
+        ones_matrix = np.ones((n_subjects, n_subjects))
+        
+    elif len(np.unique(group_assignment)) > 2:
+        raise ValueError("This test is not valid for more than "
+                         f"2 groups! (got {n_groups})")
+    else:
+        raise ValueError("Invalid group assignments!")
+    
+    # Infer subjects, groups, voxels and print for user to check
+    n_voxels = iscs.shape[1]
+    print(f"Assuming {n_subjects} subjects, {n_groups} group(s), "
+          f"and {n_voxels} voxel(s) or ROI(s).")
+    
+    # Set up permutation type (exact or Monte Carlo)
+    if n_groups == 1:
+        if n_permutations < 2**n_subjects:
+            print("One-sample approximate permutation test using sign-flipping "
+                  "procedure with Monte Carlo resampling.")
+            exact_permutations = None
+        elif n_permutations >= 2**n_subjects:
+            print("One-sample exact permutation test using sign-flipping "
+                  f"procedure with 2**{n_subjects} ({2**n_subjects}) iterations.")
+            exact_permutations = list(it.product([-1, 1], repeat=n_subjects))
+            n_permutations = 2**n_subjects
+    elif n_groups == 2:
+        if n_permutations < np.math.factorial(n_subjects):
+            print("Two-sample approximate permutation test using "
+                  "group randomization with Monte Carlo resampling.")
+            exact_permutations = None
+        elif n_permutations >= np.math.factorial(n_subjects):
+            print("Two-sample exact permutation test using group "
+                  f"randomization with {n_subjects}! "
+                  f"({np.math.factorial(n_subjects)}) "
+                  "iterations.")
+            exact_permutations = list(it.permutations(
+                np.arange(len(group_assignment))))
+            n_permutations = np.math.factorial(n_subjects)
+    
+    # If one group, just get observed summary statistic
+    if n_groups == 1:
```

Comment:
This function doesn't perform input validation for `summary_statistic` at the beginning. So I guess this `if-elif` might fail to catch invalid option for `summary_statistic`?